File: 3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i/3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i.java
class Solution {
    public int minFlips(int[][] grid) {

        int m = grid.length;
        int n = grid[0].length;

        // Rows ko palindrome banana
        int rowFlips = 0;

        for (int i = 0; i < m; i++) {

            int j = 0;
            int k = n - 1;

            while (j < k) {

                if (grid[i][j] != grid[i][k]) {
                    rowFlips++;
                }

                j++;
                k--;
            }
        }


        // Columns ko palindrome banana
        int colFlips = 0;

        for (int j = 0; j < n; j++) {

            int i = 0;
            int k = m - 1;

            while (i < k) {

                if (grid[i][j] != grid[k][j]) {
                    colFlips++;
                }

                i++;
                k--;
            }
        }

        return Math.min(rowFlips, colFlips);
    }
}

File: 3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i/3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i.cpp
class Solution {
public:
    int minFlips(vector<vector<int>>& mat) {
        int n=mat.size();
        int m=mat[0].size();
        int rowflips=0;
        for(int i=0;i<n;i++){
            int j=0;
            int k=m-1;
            while(j<k){
                if(mat[i][j]!=mat[i][k]){
                    rowflips++;
                }
                j++;
                k--;
            }
        }
        int colflips=0;
        for(int i=0;i<m;i++){
            int j=0;
            int k=n-1;
            while(j<k){
                if(mat[j][i]!=mat[k][i]){
                    colflips++;
                }
                j++;
                k--;
            }
        }
        return min(rowflips,colflips);

        
    }
};